dsmr_parser/parsers.py

```diff
@@ -84,25 +84,25 @@ def parse(self, telegram_data, encryption_key="", authentication_key="", throw_e
 
         telegram = Telegram()
 
-        for signature, parser in self.telegram_specification['objects'].items():
-            pattern = self.telegram_specification_regexes[signature]
+        for object in self.telegram_specification['objects']:
+            pattern = self.telegram_specification_regexes[object["obis_reference"]]
             matches = pattern.findall(telegram_data)
 
             # Some signatures are optional and may not be present,
             # so only parse lines that match
             for match in matches:
                 try:
-                    dsmr_object = parser.parse(match)
+                    dsmr_object = object["value_parser"].parse(match)
                 except ParseError:
-                    logger.error("ignore line with signature {}, because parsing failed.".format(signature),
+                    logger.error("ignore line with signature {}, because parsing failed.".format(object["obis_reference"]),
                                  exc_info=True)
                     if throw_ex:
                         raise
                 except Exception as err:
                     logger.error("Unexpected {}: {}".format(type(err), err))
                     raise
                 else:
-                    telegram.add(obis_reference=signature, dsmr_object=dsmr_object)
+                    telegram.add(obis_reference=object["value_name"], dsmr_object=dsmr_object)
```

Comment:
As long as this part in the Telegram class still works:
```
        # Fill dict which is only used for backwards compatibility
        if obis_reference not in self:
            self[obis_reference] = dsmr_object
```

It maintains backwards compatibility with users doing key lookups based on the regex `telegram[obis-references.P1_MESSAGE_HEADER ]`

Reply:
What do you think with the new changes?